src/stores/prompts.js

```diff
@@ -250,34 +283,89 @@ export const usePromptStore = defineStore('prompts', {
     },
 
     async addLike(id) {
-      this._prompts = []
-      this.$patch({ _prompts: this.getPrompts })
-
       this._isLoading = true
-      await updateDoc(doc(db, 'prompts', id), {
-        'info.likes': arrayUnion(useUserStore().getUserRef),
-        'info.dislikes': arrayRemove(useUserStore().getUserRef)
-      })
-        .then(() => this.fetchPromptById(id))
-        .catch((error) => {
-          console.error(error)
-          throw new Error(error)
+
+      // Export reused peace of code that create and save a prompt like
+      async function createAndSaveLike() {
+        // const likeRef= doc(db, 'prompts', id, 'opinions')
+        await setDoc(doc(db, 'prompts', id, 'opinions'), {
```

Comment:
Evaluate whether it is better to have one collection "opinions" instead of two collections ("likes" and "dislikes").
I believe that for the counts it is better to work with two.

To use `setDoc` you must define an id as last parameter.
Example:
```js
await setDoc(doc(db, 'prompts', id, 'opinions', userId), {
```